### PR TITLE
Support GIT_* vars even when a source folder was specified

### PR DESCRIPTION
**Pull request by stuarteberg**
**tl;dr**: This little PR fixes `GIT_DESCRIBE_TAG`, etc. in the case when `meta.yaml` specifies a custom `source: folder:`.

**Details:**

As described in the [docs](https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#destination-path), the `source` section can optionally specify a nested `folder` within the `work` directory for the source to be cloned into.  For example:

```yaml
source:
  git_url: http://github.com/foo/bar
  folder: foo-sources/bar
```

...but the code that sets jinja variables such as `GIT_DESCRIBE_TAG` had assumed that the recipe did not specify a custom `folder`.  This PR fixes that.

PS -- I didn't augment the test suite, partly because this is a tiny change and partly because a number of tests were already failing in my machine.  (I think there may be something wrong with my setup.)

### CODE REVIEW

File: conda_build/environ.py [hunk 1]
```diff
@@ -370,8 +370,10 @@ def meta_vars(meta, config):
         else:
             d[var_name] = value
 
-    git_dir = join(config.work_dir, '.git')
-    hg_dir = join(config.work_dir, '.hg')
+    folder = meta.get_value('source/0/folder', '')
```
Comment (msarahan) [inline conda_build/environ.py]: Is this hardcoding it to only the first entry?

Comment (stuarteberg) [reply conda_build/environ.py]: Yes, but that was already the case for `GIT_` variables -- they apply only to the first entry in the `source` list.

Comment (msarahan) [reply conda_build/environ.py]: Right.  Fair enough.  Thanks for fixing this.  Generalizing would be nice, but it's not immediately obvious to me how to do this nicely.
